lib/semantics/symbol.h

```diff
@@ -204,8 +204,13 @@ class ProcEntityDetails : public EntityDetails, public WithPassArg {
   void set_interface(const ProcInterface &interface) { interface_ = interface; }
   inline bool HasExplicitInterface() const;
 
+  std::optional<const Symbol *> init() const { return init_; }
+  void set_init(const Symbol &symbol) { init_ = &symbol; }
+  void set_init(std::nullptr_t) { init_ = nullptr; }
+
 private:
   ProcInterface interface_;
+  std::optional<const Symbol *> init_;  // if present but null => NULL()
```

Comment:
I think the comment should go with `init()` and `set_init()`. The distinction between nullopt and an optional holding nullptr is pretty subtle.

Reply:
Sure.